public/app/plugins/datasource/prometheus/metric_find_query.ts

```diff
@@ -67,7 +67,7 @@ export default class PrometheusMetricFindQuery {
 
       return this.datasource._request("GET", url).then(function(result) {
         var _labels = _.map(result.data.data, function(metric) {
-          return metric[label];
+          return metric[label] || '';
```

Comment:
should it really return an empty string in this case? Should it not return an empty array?  (ie filter out the labels)

Reply:
Should it not return an empty array.
Update the code.